### PR TITLE
Issue127 fix shuffle animation

### LINKED ISSUE #127: Array shuffle algorithm has error when items off canvas
Normally items in the array being shown in SimpleSorting and other array visualizations are always fully visible on the canvas.  When the canvas is shrunk so that some are not visible, however, a Shuffle operation can throw an error:
```
  File "SimpleSorting.py", line 734, in <lambda>
    "Shuffle", lambda: self.shuffle())
  File "SimpleSorting.py", line 397, in shuffle
    max(-DX, -bBox[0]), min(DX, canvasDimensions[0] - bBox[2]))
  File "/usr/local/Cellar/python/3.7.5/Frameworks/Python.framework/Versions/3.7/lib/python3.7/random.py", line 222, in randint
    return self.randrange(a, b+1)
  File "/usr/local/Cellar/python/3.7.5/Frameworks/Python.framework/Versions/3.7/lib/python3.7/random.py", line 200, in randrange
    raise ValueError("empty range for randrange() (%d,%d, %d)" % (istart, istop, width))
ValueError: empty range for randrange() (-30,-62, -32)
```
This is due to the algorithm attempting to keep the elements visible on the canvas throughout the shuffle. 

Another problem is that the length of the shuffle operation is currently based on number of items in the array.  With the bigger arrays in AdvancedSorting.py, that becomes a problem.

Replace the current shuffle animation with one that's of constant duration (perhaps 1 seconds at fastest speed) and doesn't produce errors.  Perhaps giving each cell a velocity vector and using repulsion/attraction to move away/towards their source/destinations would be suitable.

### PR DESCRIPTION
**Pull request by JMCanning78**
I changed the shuffle animation algorithm to a velocity-based approach to close #127.   Each cell starts with a random velocity vector that pushes it away from its starting position, down, and towards its target position (using the `random.shuffle()` permutation to determine the new position).  The velocity vector is adjusted on each animation iteration to point more and more towards the target, with some added random jitter.  The path for an individual element is sort of a spiral to its final location in the array.

Also: 
  * Prepared code for drawable's display_shape or display_val field being `None`.
  * Fixed extraction of color attribute from itemconfiguration calls.

### CODE REVIEW

**Review by lilypolonetsky — CHANGES_REQUESTED** [review]
Aside from the extra closed parentheses, this looks great! I tested out the shuffle with resized windows and there was no error.

One other thought - I think the missing [-1] after `itemconfigure(cellPair[0], 'fill')` may also be in other parts of the code (since much was based off of SimpleSorting.py). Should we open an issue to deal with this?

File: PythonVisualizations/SimpleSorting.py [hunk 1]
```diff
@@ -369,43 +369,47 @@ def find(self, val):
...
+             random.randint(-1, int(dx / 3) + 1),
+             random.randint(int(height * 1 / 3), height + int(abs(dx) / 8)))
+            for dx, dy in [self.toTarget(i) for i in range(nItems)]]
+        
+        steps = 20
+        for step in range(steps):
+            stepsToGo = steps - step
+            jitter = max(1, int(stepsToGo * 2 / 3))
+            half = max(1, jitter // 2)
+            for i in range(nItems)):
```
[the remark this answers is not in the document]
Comment (JMCanning78) [reply PythonVisualizations/SimpleSorting.py]: I fixed the extra parenthesis on line 399 in commit c5ba1e2 .  I couldn't find any calls to `itemconfigure()` that were missing a [-1] to extract the last item, so I skipped that.  I did merge in the changes from the `master` branch to resolve the one conflict that had arose.

**Comment by JMCanning78** [comment]
I think I fixed the problems mentioned in [your comment](https://github.com/JMCanning78/datastructures-visualization/pull/128#pullrequestreview-511267385), Lily.

**Review by lilypolonetsky — APPROVED** [review]
Looks great!

**Pull request merged**